docs/developer/release-team.md

```diff
@@ -0,0 +1,80 @@
+# Cluster API Release Team
+
+## Overview
+
+In the past, releasing Cluster API has been mostly ad-hoc and relied on one or more contributors to do most of the chore work necessary to prepare the release. One of the major downsides of this approach is that it is often difficult for users and providers to plan around Cluster API releases as they often have little visibility around when a release will happen. 
+
+This document introduces the concept of a release team with the following goals and non-goals:
+
+### Goals
+
+- To improve communication to end users and contributors about CAPI release cadence and target release dates
+- To spread the load of release tasks and involve a bigger, more diverse set of CAPI contributors in cutting releases
+- To look at the Kubernetes SIG release team processes for guidance on releasing best practices
+- To improve tooling, documentation, and automation of the CAPI release process
+
+### Non-Goals/Future work
+
+- To increase the frequency of releases (AKA release cadence). This will be revisited in the future once the release process stabilizes.
+- To change the current proposal (CAEP) process
+- To copy implement all steps of the Kubernetes release process for CAPI releases
+
+Note that this document is intended to be a starting point for the release team. It is not a complete release process document.
+
+More details on the CAPI release process can be found in [this past issue tracking release tasks](https://github.com/kubernetes-sigs/cluster-api/issues/6615) and the current [release documentation](https://github.com/kubernetes-sigs/cluster-api/blob/main/docs/developer/releasing.md).
+
+## Duration of Term
+
+Each release team term will last approximately four months, to align with one minor release cycle. A minor release cycle starts right after a minor release and concludes with the release of the next minor release. There is no limit to the number of terms a release team member can serve, meaning that it's possible for a release team member to serve multiple consecutive terms.
+
+As noted above, making changes to  the CAPI release cadence is out of scope for this initial release team process. 
+
+## Specific Responsibilities
+
+- Release patch releases monthly or more often as needed, so users will get fixes & updated dependencies with CVE fix on a predictable cadence
+- Release a minor release every 4 months (to be revised, see future work), so users will get new features on a predictable cadence
+- Create beta and rc tags for upcoming minor releases
+- Ensure eligible PRs are cherry-picked to the active release branches
```

Comment:
No strong opinions, but to ensure this the release teams has to be aware of all merged PRs.

I wonder if that should be the responsibility of PR reviewers and especially approvers instead. I can imagine this would scale better as PR reviewers/approvers are already aware of PRs and their content while release team members don't necessarily watch all PRs (? or maybe they do)

Reply:
I reworded to "Ensure _only_ eligible PRs are cherry-picked", does that help? I agree that the release team shouldn't be aware of every merge PR but I do think they should be aware of all cherry-pick PRs. That way, PR reviewers can submit a PR for cherry-pick and release team can approve the cherry pick (we don't have a formal process of this at the moment, could be as simple as having release team member lgtm'ing the PR).

Reply:
Sounds good to me!